mission_control/test/test_if_mission_control_aborts_when_health_monitor_reports_fault.py

```diff
@@ -0,0 +1,141 @@
+        self.mission_to_load.mission_file_full_path = self.dir_path + \
+            "test_missions/mission.xml"
+        self.mission_to_execute = ExecuteMission()
+        self.mission_to_execute.mission_id = 1
+        self.simulate_error_code = ReportFault()
+        self.simulate_error_code.fault_id = ReportFault.PAYLOAD_ERROR
+        self.mission_load_state = ReportLoadMissionState()
+        self.report_execute_mission = ReportExecuteMissionState()
+
+        # Subscribers
+        self.exec_state_sub = rospy.Subscriber('/mission_control_node/report_mission_execute_state',
+                                               ReportExecuteMissionState, self.callback_mission_execute_state)
+
+        self.load_state_sub = rospy.Subscriber('/mission_control_node/report_mission_load_state',
+                                               ReportLoadMissionState, self.callback_mission_load_state)
+
+        #   Publisher
+
+        self.simulated_mission_control_load_mission_pub = rospy.Publisher('/mission_control_node/load_mission',
+                                                                          LoadMission, latch=True, queue_size=1)
+
+        self.simulated_mission_control_execute_mission_pub = rospy.Publisher(
+            '/mission_control_node/execute_mission', ExecuteMission, latch=True, queue_size=1)
+
+        self.simulated_health_monitor_pub = rospy.Publisher('/health_monitor/report_fault',
+                                                            ReportFault, queue_size=1)
+
+    def callback_mission_execute_state(self, msg):
+        self.report_execute_mission = msg
+
+    def callback_mission_load_state(self, msg):
+        self.mission_load_state = msg.load_state
+        self.mission_load_state_flag = True
+
+    def test_mission_control_aborts_if_health_monitor_reports_fault(self):
+
+        # Load Mission
+        self.simulated_mission_control_load_mission_pub.publish(
+            self.mission_to_load)
+
+        def load_valid_mission():
+            return self.mission_load_state == ReportLoadMissionState.SUCCESS
+        self.assertTrue(wait_for(load_valid_mission),
+                        msg='Error Loading mission')
+
+        # Execute Mission and check if the mission control reports status
+        self.simulated_mission_control_execute_mission_pub.publish(
+            self.mission_to_execute)
+
+        def success_mission_status_is_reported():
+            return self.report_execute_mission.execute_mission_state == ReportExecuteMissionState.EXECUTING
+        self.assertTrue(wait_for(success_mission_status_is_reported),
+                        msg='Mission control must report SUCCESS')
+
+        # Simulate the health monitor publishing the fault code
+        self.simulated_health_monitor_pub.publish(self.simulate_error_code)
+
+        def abort_mission_status_is_reported():
+            return self.report_execute_mission.execute_mission_state == ReportExecuteMissionState.PAUSED
```

Comment:
@patapufate `abort` is `paused`? I'm confused.

Reply:
because the action didn't set a Failure state. The fault was sent by health monitor, so we are stopping the action. I'm modifying the test to be more robust.

Reply:
> because the action didn't set a Failure state.

I think we're conflating mission state with behavior tree state. It's odd to have a mission switch to PAUSED on abort. It's not paused. You cannot resume it. It's aborted.